presto-docs/src/main/sphinx/functions/geospatial.rst

```diff
@@ -0,0 +1,191 @@
+=====================
+GeoSpatial Functions
+=====================
+
+Presto GeoSpatial functions support SQL/MM specification
+
+Presto GeoSpatial functions are compliant with the Open Geospatial Consortium’s (OGC) OpenGIS Specifications. As such, many Presto GeoSpatial functions require, or more accurately, assume that geometries that are operated on are both simple and valid. For example, it does not make sense to calculate the area of a polygon that has a hole defined outside of the polygon, or to construct a polygon from a non-simple boundary line.
+
+Presto GeoSpatial functions support the Well-Known Text (WKT) form of spatial objects. Here are the examples:
+
+POINT (0 0)
+
+LINESTRING (0 0, 1 1, 1 2)
+
+POLYGON ((0 0, 4 0, 4 4, 0 4, 0 0), (1 1, 2 1, 2 2, 1 2, 1 1))
+
+MULTIPOINT (0 0, 1 2)
+
+MULTILINESTRING ((0 0, 1 1, 1 2), (2 3, 3 2, 5 4))
+
+MULTIPOLYGON (((0 0, 4 0, 4 4, 0 4, 0 0), (1 1, 2 1, 2 2, 1 2, 1 1)), ((-1 -1, -1 -2, -2 -2, -2 -1, -1 -1)))
+
+GEOMETRYCOLLECTION (POINT(2 3), LINESTRING (2 3, 3 4))
+
+Constructors
+------------
+
+.. function:: ST_Point(double, double) -> point
+
+    Returns a Geometry type Point object with the given coordinate values.
+
+.. function:: ST_LineFromText(varchar) -> line
+
+    Returns a Geometry type LineString object from Well-Known Text representation (WKT).
+
+.. function:: ST_Polygon(varchar) -> polygon
+
+    Returns a Geometry type Polygon object from Well-Known Text representation (WKT).
+
+.. function:: ST_GeometryFromText(varchar) -> Geometry
+
+    Returns a Geometry type object from Well-Known Text representation (WKT).
+
+.. function:: stAsText(Geometry) -> varchar
+
+    Returns the Well-Known Text (WKT) representation of the Geometry. For empty Geometries, ST_AsText(ST_LineFromText('LINESTRING EMPTY')) will produce 'MULTILINESTRING EMPTY', ST_AsText(ST_Polygon('POLYGON EMPTY')) will produce 'MULTIPOLYGON EMPTY'.
+
+Relationship Tests
+------------------
+
+.. function:: ST_Contains(Geometry, Geometry) -> boolean
+
+    Returns TRUE if and only if no points of right lie in the exterior of left, and at least one point of the interior of left lies in the interior of right.
+
+.. function:: ST_Crosses(Geometry, Geometry) -> boolean
+
+    Returns TRUE if the supplied geometries have some, but not all, interior points in common.
+
+.. function:: ST_Disjoint(Geometry, Geometry) -> boolean
+
+    Returns TRUE if the Geometries do not "spatially intersect" - if they do not share any space together.
```

Comment:
`given geometries` (as you used in the function below, also doesn't make sense to capitalize `Geometries`).